sanic_routing/route.py

```diff
@@ -152,6 +167,9 @@ def add_parameter(
             cast,
             pattern,
             label not in self.router.regex_types,
+            list(self.router.regex_types.keys()).index(label)
+            if label in self.router.regex_types
+            else 0,
```

Comment:
What is at index 0 of that map? String type, presumably, but a comment here would be helpful.

Reply:
I think if the code is reworked slightly it becomes self-documenting:

```python
        is_regex = label not in self.router.regex_types
        priority = (
            0
            if is_regex
            else list(self.router.regex_types.keys()).index(label)
        )
        self._params[idx] = ParamInfo(
            name, raw_path, label, cast, pattern, is_regex, priority
        )
```